### PR TITLE
NetSim: Style cleanup

### PR DESCRIPTION
**Pull request by islemaster**
Some of Brent's comments on https://github.com/code-dot-org/code-dot-org/pull/4081 convinced me that I'm falling down on the job :smiley: so here are some improvements to code style in NetSim:
- No more trailing parentheses on `describe`, `it`, or `beforeEach` [per our style guide regarding function invocation](https://github.com/code-dot-org/code-dot-org/blob/staging/STYLEGUIDE.md#js-parentheses-spacing).
- Prefer [Chai's assertion library](http://chaijs.com/api/assert/) over some custom assertion types we were using.
- Enable the [eqeqeq JSHint option](http://jshint.com/docs/options/#eqeqeq) on NetSim files to enforce use of strict equality (which caught a few places we'd missed this).
- Enable strict linting in NetSim test files, similar to the strict linting used in other NetSim source.

Still true:

```
╰○ grunt test
Running "jshint:all" (jshint) task
>> 456 files lint free.

Running "exec:mochaTest" (exec) task
[...test output...]
  1224 passing (2m)
```

### CODE REVIEW

File: apps/test/netsim/NetSimLogger.js [hunk 1]
```diff
@@ -15,8 +26,8 @@ var OldConsole = function () {
 };
 
 // Modern console with log, info, warn, error methods.
-var ModernConsole = function () {
-  var that = OldConsole();
```
Comment (Hamms) [inline apps/test/netsim/NetSimLogger.js]: :heart: parasitic inheritance

File: apps/test/netsim/NetSimTable.js [hunk 2]
```diff
@@ -300,247 +309,248 @@ describe("NetSimTable", function () {
...
       });
     });
+    /* jshint unused: true */
   });
 
-  it ("calls readAll on the API table", function () {
+  it("calls readAll on the API table", function () {
     netsimTable.refresh(callback);
-    assertEqual(apiTable.log(), 'readAll');
+    assert.equal(apiTable.log(), 'readAll');
```
Comment (Hamms) [inline apps/test/netsim/NetSimTable.js]: Most of the rest of the `assert.equal` calls follow the `assert.equal(expected, actual)` pattern. This one reverses that.

However, according to the [chai documentation](http://chaijs.com/api/assert/) this is the preferred order, so I'm not sure what we want to do about that.

File: apps/test/netsim/NetSimVizWire.js [hunk 3]
```diff
@@ -26,33 +28,33 @@ describe("NetSimVizWire", function () {
...
 
-    it ("has default properties", function () {
-      assertEqual(0, vizWire.textPosX_);
-      assertEqual(0, vizWire.textPosY_);
-      assertEqual([], vizWire.encodings_);
+    it("has default properties", function () {
+      assert.equal(0, vizWire.textPosX_);
+      assert.equal(0, vizWire.textPosY_);
+      assert.deepEqual([], vizWire.encodings_);
       assert(localVizNode === vizWire.localVizNode);
```
Comment (Hamms) [inline apps/test/netsim/NetSimVizWire.js]: Any reason we're not doing `assert.strictEqual` here?

File: apps/test/netsim/NetSimWire.js [hunk 4]
```diff
@@ -16,36 +26,36 @@ describe("NetSimWire", function () {
     wireTable = testShard.wireTable;
   });
 
-  it ("uses the wire table", function () {
+  it("uses the wire table", function () {
     var wire = new NetSimWire(testShard);
     assert(wire.getTable() === testShard.wireTable);
```
Comment (Hamms) [inline apps/test/netsim/NetSimWire.js]: this looks like an appropriate place for `assert.deepEqual`

File: apps/test/netsim/dataConverters.js [hunk 5]
```diff
@@ -708,31 +715,31 @@ describe("DataConverters", function () {
      * @param {Base64Payload} right
      */
     var assertBase64NotEqual = function (left, right) {
-      var stringsEqual = (left.string == right.string);
-      var lensEqual = (left.len == right.len);
+      var stringsEqual = (left.string === right.string);
+      var lensEqual = (left.len === right.len);
       assert(!(stringsEqual && lensEqual), "Expected " + JSON.stringify(left) + " and " +
         JSON.stringify(right) + " to be not equal, but they are");
```
Comment (Hamms) [inline apps/test/netsim/dataConverters.js]: Why not just use two asserts?

File: apps/test/util/testUtils.js [hunk 6]
```diff
@@ -132,65 +132,6 @@ exports.runOnAppTick = function (app, tick, fn) {
 };
 
 /**
- * Deep equality check of two values with more useful assertion failure
- * message.  Depends on lodash isEqual.
- * @param {*} left
- * @param {*} right
- */
-exports.assertEqual = function (left, right) {
```
Comment (Hamms) [inline apps/test/util/testUtils.js]: this and `assertThrows` are also used by `apps/test/commandsTest.js`

**Comment by Hamms** [comment]
It might be worth doing a `cd apps/test/netsim; git grep "assert("` and glancing through all the places where we're still calling `assert()` directly, just to make sure none of them should be some other call

**Comment by Hamms** [comment]
LGTM after comments!

File: apps/test/netsim/NetSimLogger.js (hunk 1 above)
Comment (islemaster) [inline apps/test/netsim/NetSimLogger.js]: :stuck_out_tongue_closed_eyes: Any action you want taken here?  I know it's clumsy, but it's just for a tiny sensing interface.

File: apps/test/netsim/NetSimTable.js (hunk 2 above)
Comment (islemaster) [inline apps/test/netsim/NetSimTable.js]: Aww crud.  Now I have to go fix them all.

File: apps/test/netsim/NetSimLogger.js (hunk 1 above)
Comment (Hamms) [inline apps/test/netsim/NetSimLogger.js]: Heh, that wasn't sarcasm. I legitimately do love parasitic inheritance. I had no reason beyond that for calling it out

File: apps/test/netsim/NetSimVizWire.js (hunk 3 above)
Comment (islemaster) [inline apps/test/netsim/NetSimVizWire.js]: Only a lazy replacement.  Do you think we should be using strictEqual as much as possible, or only in places like this where we might confuse an `undefined` for a zero?

File: apps/test/netsim/NetSimTable.js (hunk 2 above)
Comment (Hamms) [inline apps/test/netsim/NetSimTable.js]: It's a lot of work for a very minor nit

File: apps/test/netsim/dataConverters.js (hunk 5 above)
Comment (islemaster) [inline apps/test/netsim/dataConverters.js]: Better yet:

```javascript
      assert.notDeepEqual(left, right, "Expected " + JSON.stringify(left) + " and " +
        JSON.stringify(right) + " to be not equal, but they are");
```

File: apps/test/netsim/NetSimVizWire.js (hunk 3 above)
Comment (Hamms) [inline apps/test/netsim/NetSimVizWire.js]: I was specifically referrng to the `assert(localVizNode === vizWire.localVizNode)` call becoming `assert.strictEqual(localVizNode, vizWire.localVizNode)`

File: apps/test/util/testUtils.js (hunk 6 above)
Comment (islemaster) [inline apps/test/util/testUtils.js]: I actually touched those files too (I think NetSim might be the only client for the commands stuff).

File: apps/test/netsim/NetSimVizWire.js (hunk 3 above)
Comment (Hamms) [inline apps/test/netsim/NetSimVizWire.js]: I suppose there's also the question of whether `.strictEqual` or `.deepEqual` is more appropriate here

File: apps/test/netsim/dataConverters.js (hunk 5 above)
Comment (Hamms) [inline apps/test/netsim/dataConverters.js]: or removing  this helper entirely and replacing it with calls to `assert.notDeepEqual`

File: apps/test/netsim/NetSimWire.js (hunk 4 above)
Comment (islemaster) [inline apps/test/netsim/NetSimWire.js]: :+1:

File: apps/test/util/testUtils.js (hunk 6 above)
Comment (Hamms) [inline apps/test/util/testUtils.js]: so you did! My mistake

File: apps/test/netsim/NetSimVizWire.js (hunk 3 above)
Comment (islemaster) [inline apps/test/netsim/NetSimVizWire.js]: Oh!  I see.  `strictEqual` is more consistent with the original test, as we're checking object identity.

File: apps/test/netsim/NetSimTable.js (hunk 2 above)
Comment (islemaster) [inline apps/test/netsim/NetSimTable.js]: Yeah, too much.  But we can move toward that standard.  I'll make sure tests I touch from here on out get corrected.

**Comment by islemaster** [comment]
Did another pass through, picking more precise Chai assertions where it seemed potentially helpful.

**Pull request merged**